From cea19fed19f0968f2458bac329b2506d4a0e8452 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rn=20Kottmann?= <joern@apache.org>
Date: Tue, 5 Apr 2011 11:10:54 +0000
Subject: [PATCH] OPENNLP-150 Added the missing language parameter to the
 training descriptors. Thanks to Tommaso Teofili for providing the patch and
 for testing the UIMA integration.

git-svn-id: https://svn.apache.org/repos/asf/incubator/opennlp/trunk@1088972 13f79535-47bb-0310-9956-ffa450edef68
---
 opennlp-uima/descriptors/ChunkerTrainer.xml     | 16 ++++++++++++++++
 .../descriptors/PersonNameFinderTrainer.xml     | 17 ++++++++++++++++-
 opennlp-uima/descriptors/PosTaggerTrainer.xml   | 16 ++++++++++++++++
 .../descriptors/SentenceDetectorTrainer.xml     | 16 ++++++++++++++++
 opennlp-uima/descriptors/TokenizerTrainer.xml   | 12 ++++++++++++
 5 files changed, 76 insertions(+), 1 deletion(-)

diff --git a/opennlp-uima/descriptors/ChunkerTrainer.xml b/opennlp-uima/descriptors/ChunkerTrainer.xml
index 01025b240..addaf6a16 100644
--- a/opennlp-uima/descriptors/ChunkerTrainer.xml
+++ b/opennlp-uima/descriptors/ChunkerTrainer.xml
@@ -56,6 +56,14 @@
 				<multiValued>false</multiValued>
 				<mandatory>true</mandatory>
 			</configurationParameter>
+
+			<configurationParameter>
+				<name>opennlp.uima.Language</name>
+				<type>String</type>
+				<multiValued>false</multiValued>
+				<mandatory>true</mandatory>
+			</configurationParameter>
+
 		</configurationParameters>
 
 		<configurationParameterSettings>
@@ -86,6 +94,14 @@
 					<string>pos</string>
 				</value>
 			</nameValuePair>
+
+			<nameValuePair>
+				<name>opennlp.uima.Language</name>
+				<value>
+					<string>en</string>
+				</value>
+			</nameValuePair>
+
 		</configurationParameterSettings>
 		
 		<typeSystemDescription />
diff --git a/opennlp-uima/descriptors/PersonNameFinderTrainer.xml b/opennlp-uima/descriptors/PersonNameFinderTrainer.xml
index c6013c397..e7215246f 100644
--- a/opennlp-uima/descriptors/PersonNameFinderTrainer.xml
+++ b/opennlp-uima/descriptors/PersonNameFinderTrainer.xml
@@ -71,7 +71,14 @@
 				<mandatory>false</mandatory>
 			</configurationParameter>
 		</configurationParameters>
-		
+
+			<configurationParameter>
+				<name>opennlp.uima.Language</name>
+				<type>String</type>
+				<multiValued>false</multiValued>
+				<mandatory>true</mandatory>
+			</configurationParameter>
+
 		<configurationParameterSettings>
 
 			<nameValuePair>
@@ -101,6 +108,14 @@
 					<string>opennlp.uima.Person</string>
 				</value>
 			</nameValuePair>
+
+			<nameValuePair>
+				<name>opennlp.uima.Language</name>
+				<value>
+					<string>en</string>
+				</value>
+			</nameValuePair>
+
 		</configurationParameterSettings>
 		<typeSystemDescription>
 			<imports>
diff --git a/opennlp-uima/descriptors/PosTaggerTrainer.xml b/opennlp-uima/descriptors/PosTaggerTrainer.xml
index 027c7a406..516805e96 100644
--- a/opennlp-uima/descriptors/PosTaggerTrainer.xml
+++ b/opennlp-uima/descriptors/PosTaggerTrainer.xml
@@ -56,6 +56,14 @@
 				<multiValued>false</multiValued>
 				<mandatory>true</mandatory>
 			</configurationParameter>
+
+			<configurationParameter>
+				<name>opennlp.uima.Language</name>
+				<type>String</type>
+				<multiValued>false</multiValued>
+				<mandatory>true</mandatory>
+			</configurationParameter>
+
 		</configurationParameters>
 
 		<configurationParameterSettings>
@@ -86,6 +94,14 @@
 					<string>pos</string>
 				</value>
 			</nameValuePair>
+
+			<nameValuePair>
+				<name>opennlp.uima.Language</name>
+				<value>
+					<string>en</string>
+				</value>
+			</nameValuePair>
+
 		</configurationParameterSettings>
 		
 		<typeSystemDescription />
diff --git a/opennlp-uima/descriptors/SentenceDetectorTrainer.xml b/opennlp-uima/descriptors/SentenceDetectorTrainer.xml
index e05acc174..568776334 100644
--- a/opennlp-uima/descriptors/SentenceDetectorTrainer.xml
+++ b/opennlp-uima/descriptors/SentenceDetectorTrainer.xml
@@ -43,6 +43,14 @@
 				<multiValued>false</multiValued>
 				<mandatory>true</mandatory>
 			</configurationParameter>
+
+			<configurationParameter>
+				<name>opennlp.uima.Language</name>
+				<type>String</type>
+				<multiValued>false</multiValued>
+				<mandatory>true</mandatory>
+			</configurationParameter>
+
 		</configurationParameters>
 
 		<configurationParameterSettings>
@@ -59,6 +67,14 @@
 					<string>opennlp.uima.Sentence</string>
 				</value>
 			</nameValuePair>
+
+			<nameValuePair>
+				<name>opennlp.uima.Language</name>
+				<value>
+					<string>en</string>
+				</value>
+			</nameValuePair>
+
 		</configurationParameterSettings>
 		<typeSystemDescription />
 		<typePriorities />
diff --git a/opennlp-uima/descriptors/TokenizerTrainer.xml b/opennlp-uima/descriptors/TokenizerTrainer.xml
index 4d01014d5..a1463db4a 100644
--- a/opennlp-uima/descriptors/TokenizerTrainer.xml
+++ b/opennlp-uima/descriptors/TokenizerTrainer.xml
@@ -55,6 +55,12 @@
 				<multiValued>false</multiValued>
 				<mandatory>true</mandatory>
 			</configurationParameter>
+			<configurationParameter>
+				<name>opennlp.uima.Language</name>
+				<type>String</type>
+				<multiValued>false</multiValued>
+				<mandatory>true</mandatory>
+			</configurationParameter>
 		</configurationParameters>
 		<configurationParameterSettings>
 			<nameValuePair>
@@ -83,6 +89,12 @@
 					<boolean>false</boolean>
 				</value>
 			</nameValuePair>
+			<nameValuePair>
+				<name>opennlp.uima.Language</name>
+				<value>
+					<string>en</string>
+				</value>
+			</nameValuePair>
 		</configurationParameterSettings>
 		<typeSystemDescription>
 			<imports>
